### PR TITLE
Convert Zebra device to ophyd async

### LINKED ISSUE #356: Ophyd-async: Convert the `zebra` device to ophyd-async
Convert the Zebra device to ophyd-async.

## Acceptance Criteria
- Tests still pass both here and in hyperion/mx-bluesky
- `Zebra` uses ophyd-async

### PR DESCRIPTION
**Pull request by noemifrisina**
Fixes #356

### Instructions to reviewer on how to test:
1. Tests pass
2. Plans using the zebra device in Hyperion and mx_bluesky are updated where needed
3. Tests still pass in Hyperion and mx_bluesky

### Checks for reviewer
- [ ] Would the PR title make sense to a scientist on a set of release notes
- [ ] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)

### CODE REVIEW

File: src/dodal/devices/zebra.py [hunk 1]
```diff
@@ -74,70 +80,85 @@ class FastShutterAction(IntEnum):
...
 
     TIMEOUT = 3
 
-    arm_set = Component(EpicsSignal, "PC_ARM")
-    disarm_set = Component(EpicsSignal, "PC_DISARM")
-    armed = Component(EpicsSignal, "PC_ARM_OUT")
+    def __init__(self, prefix: str, name: str = "") -> None:
+        self.arm_set = epics_signal_rw(int, prefix + "PC_ARM")
+        self.disarm_set = epics_signal_rw(int, prefix + "PC_DISARM")
+        self.armed = epics_signal_rw(ArmDemand, prefix + "PC_ARM_OUT")
```
[the remark this answers is not in the document]
Comment (noemifrisina) [reply src/dodal/devices/zebra.py:97]: Thanks, I didn't know that and the unit tests do not fail whatever the enum signal passed (opening an issue on that).

I've now turned the relevant enums in here to (str, Enum). 
Tbh, I don't really like using a boolean in this case - the arming device- , I find it clearer to have a way to ask for arm/disarm. I fixed the type of that particular signal but left the demand to be passed to the set method as `ArmDemand`.

File: src/dodal/devices/zebra.py [hunk 2]
```diff
@@ -45,99 +33,125 @@
...
+    def __init__(self, prefix: str, name: str = "") -> None:
+        self.arm_set = epics_signal_rw(float, prefix + "PC_ARM")
+        self.disarm_set = epics_signal_rw(float, prefix + "PC_DISARM")
+        self.armed = epics_signal_rw(float, prefix + "PC_ARM_OUT")
+        super().__init__(name)
 
-    def set(self, demand: ArmDemand) -> StatusBase:
-        status = await_value(self.armed, demand.value, timeout=self.TIMEOUT)
+    async def _set_armed(self, demand: ArmDemand):
+        await self.armed.set(demand.value)
```
Comment (DominicOram) [inline src/dodal/devices/zebra.py:97]: Should: I think `armed` is a read-only signal from EPICS as to whether the zebra thinks it's armed. We can't set it. Instead I think we want to observe it after we've done the set and wait for it to be as we expect. See other places we use `observe_value`

**Review by DominicOram — CHANGES_REQUESTED** [review]
Thanks, one comment, otherwise I think it's good!

**Review by olliesilvester — APPROVED** [review]
David's change looks good to me

**Pull request merged**